device/common/include/traccc/finding/device/impl/build_tracks.ipp

```diff
@@ -105,8 +102,8 @@ TRACCC_DEVICE inline void build_tracks(const global_index_t globalIndex,
         for (unsigned int i = 0; i < cands_per_track.size(); ++i) {
             for (unsigned int j = 0; j < cands_per_track.size(); ++j) {
                 if (i != j) {
-                    assert(cands_per_track.at(i).measurement_id !=
-                           cands_per_track.at(j).measurement_id);
+                    // assert(cands_per_track.at(i).measurement_id !=
+                    //       cands_per_track.at(j).measurement_id);
```

Comment:
Is this test meant to be revived in some way later on? If not, let's remove the whole thing.

While the compilers were probably smart enough to completely remove this double for-loop from the code in optimized code already, such tests would generally be nicer to do through some lambda. With the entire body of the code sitting inside of the `assert(...)` parentheses.

But for now just removing this entire block would be fine. 🤔